dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletionWithData/AzureChatCompletionWithData.cs

```diff
@@ -0,0 +1,345 @@
+
+        this._httpClient = httpClient ?? new HttpClient(NonDisposableHttpClientHandler.Instance, disposeHandler: false);
+        this._logger = logger ?? NullLogger.Instance;
+    }
+
+    /// <inheritdoc/>
+    public ChatHistory CreateNewChat(string? instructions = null)
+    {
+        return new OpenAIChatHistory(instructions);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IReadOnlyList<IChatResult>> GetChatCompletionsAsync(
+        ChatHistory chat,
+        ChatRequestSettings? requestSettings = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(chat);
+
+        requestSettings ??= new();
+
+        ValidateMaxTokens(requestSettings.MaxTokens);
+
+        return await this.ExecuteCompletionRequestAsync(chat, requestSettings, cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public IAsyncEnumerable<IChatStreamingResult> GetStreamingChatCompletionsAsync(
+        ChatHistory chat,
+        ChatRequestSettings? requestSettings = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(chat);
+
+        requestSettings ??= new();
+
+        ValidateMaxTokens(requestSettings.MaxTokens);
+
+        return this.ExecuteCompletionStreamingRequestAsync(chat, requestSettings, cancellationToken);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IReadOnlyList<ITextResult>> GetCompletionsAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        CancellationToken cancellationToken = default)
+    {
+        requestSettings ??= new();
+
+        var chat = this.PrepareChatHistory(text, requestSettings);
+        var chatRequestSettings = this.PrepareChatRequestSettings(requestSettings);
+
+        return (await this.GetChatCompletionsAsync(chat, chatRequestSettings, cancellationToken).ConfigureAwait(false))
+            .OfType<ITextResult>()
+            .ToList();
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<ITextStreamingResult> GetStreamingCompletionsAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        requestSettings ??= new();
+
+        var chat = this.PrepareChatHistory(text, requestSettings);
+        var chatRequestSettings = this.PrepareChatRequestSettings(requestSettings);
+
+        await foreach (var result in this.GetStreamingChatCompletionsAsync(chat, chatRequestSettings, cancellationToken))
+        {
+            yield return (ITextStreamingResult)result;
+        }
+    }
+
+    #region private ================================================================================
+
+    private const string ServerEventPayloadPrefix = "data:";
+    private const string DefaultApiVersion = "2023-06-01-preview";
+    private const string EndpointUriFormat = "{0}/openai/deployments/{1}/extensions/chat/completions?api-version={2}";
```

Comment:
nit point to think about, and not related to the PR: In most cases, we create constants at the class level by declaring and initializing them as **private** const class fields. One of the reasons for doing this could be to avoid "magic numbers" - _literals used in the code without any explanation of their meaning_ - by giving them the right name and making them constants. However, if that is the reason, why do we declare the constants at the class level and not at the method level? Considering that almost always the constants are used by only one method of the class, as long as they are named properly, declaring them inside the method that needs them will improve the readability of the method. Developers can have their values in front of their eyes within the method that uses them, and they won't need to go back and forth between the method code and constant declarations to access the const values. There could be a performance aspect to consider, but I think it can be negligible as .NET runtimes are very sophisticated and can optimize method-scoped constants.